### PR TITLE
enables hiding sensitivity mismatch warning in response.get_evalresp_response

### PR DESCRIPTION
**Pull request by rizac**
### What does this PR do?

This branch simply adds the possibility to hide  sensitivity mismatch warnings in response.get_evalresp_response, leaving the default of False for legacy code using the method

### Why was it initiated?  Any relevant Issues?

In the Response object public methods a user can not ignore sensitivity mismatch warning, printing to stderr and particularly tedious because the warnings seem to be issued from external C-libraries and thus they are hard (albeit not impossible) to capture from within Python. The fix is simple: the parameter hide_sensitivity_warnings is already implemented in the private class methods: This PR simply adds the optional parameter in response.get_evalresp_response, passing then the value to the nested private methods

Note: I did not implement any new test because it would involve testing external libraries which should not be ObsPy goal (on a terminal it works though)

### PR Checklist
- [x] Correct base branch selected? `master` for new features, `maintenance_...` for bug fixes
- [x] This PR is not directly related to an existing issue (which has no PR yet).
- [ ] If the PR is making changes to documentation, docs pages can be built automatically.
      Just remove the space in the following string after the + sign: "+ DOCS"
- [ ] If any network modules should be tested for the PR, add them as a comma separated list
      (e.g. `clients.fdsn,clients.arclink`) after the colon in the following magic string: "+TESTS:"
      (you can also add "ALL" to just simply run all tests across all modules)
- [x] All tests still pass.
- [ ] Any new features or fixed regressions are be covered via new tests.
- [ ] Any new or changed features have are fully documented.
- [ ] Significant changes have been added to `CHANGELOG.txt` .
- [ ] First time contributors have added your name to `CONTRIBUTORS.txt` .

### CODE REVIEW

**Comment by megies** [comment]
I'm a bit torn about this one. I understand that people might want to hide those warnings, especially when processing large amounts of data, with metadata from a wide variety of sources/authors.

However, I tend to think this should be handled in user code using e.g. `with warnings.catch_warnings(...)` context.

I'd even maybe prefer some mechanism maybe in `obspy.util` to globally deactivate specific warnings before running obpsy code.

Other opinions?

**Comment by krischer** [comment]
> I'm a bit torn about this one. I understand that people might want to hide those warnings, especially when processing large amounts of data, with metadata from a wide variety of sources/authors.
> 
> However, I tend to think this should be handled in user code using e.g. `with warnings.catch_warnings(...)` context.
> 
> I'd even maybe prefer some mechanism maybe in `obspy.util` to globally deactivate specific warnings before running obpsy code.
> 
> Other opinions?

In general I very much share this opinion - I think we can just always defer people to use Python's `warnings` module - they can filter and hide to their hearts' content.

This particular case here is different though because it must be passed on to the underlying C `evalresp` routine which either prints it or not so catching it on the python side is really hard. We added this option at one point to be able to recompute the total sensitivity without that warning but never exposed it elsewhere.

**Comment by megies** [comment]
Oh ok, I missed that point, skimming the PR.. fair enough then

File: obspy/core/inventory/response.py [hunk 1]
```diff
@@ -1620,16 +1621,22 @@ def get_evalresp_response_for_frequencies(
         :type end_stage: int, optional
         :param end_stage: Stage sequence number of last stage that will be
             used (disregarding all later stages).
+        :type hide_sensitivity_mismatch_warning: bool
+        :param hide_sensitivity_mismatch_warning: Hide the evalresp warning
+            that computed and reported sensitivities don't match.
```
Comment (megies) [inline obspy/core/inventory/response.py:1626]:
```suggestion
            that computed and reported sensitivities do not match.
```

File: obspy/core/inventory/response.py [hunk 2]
```diff
@@ -1654,6 +1661,9 @@ def get_evalresp_response(self, t_samp, nfft, output="VEL",
...
```
Comment (megies) [inline obspy/core/inventory/response.py:1666]:
```suggestion
            that computed and reported sensitivities do not match.
```

**Comment by megies** [comment]
I guess you can directly pass this down through `**kwargs` all the way from `Stream.remove_response()`? If yes maybe add a short comment in that docstring

File: obspy/core/inventory/response.py (hunk 1 above)
Comment (rizac) [reply obspy/core/inventory/response.py:1626]: Note that I copied the comment from legacy code so the "don't" is still around

File: obspy/core/inventory/response.py (hunk 2 above)
Comment (rizac) [reply obspy/core/inventory/response.py:1666]: Just a side note: I copied the comment from legacy code so the "don't" is still around

**Comment by rizac** [comment]
> I guess you can directly pass this down through `**kwargs` all the way from `Stream.remove_response()`? If yes maybe add a short comment in that docstring

I personally prefer to be explicit (which incidentally is also one of the - probably overused - Python dogmas ), given that I do not see major drawbacks (except that yes, maybe the parameter name is too long). so I would let the code as it is, if you do not mind

**Comment by rizac** [comment]
Just a side note (to reply once to all the comments in the thread). First of all, we use ObsPy as library to process large amount of data, so we absolutely need to control the output in our software. A terminal (or log file) polluted with, say, 10,000 printouts of the same line of text is obviously more than a nuisance.

Now, with that in mind, the solution proposed above to deactivate globally specific (when not all) warnings is quite catchy, and it would be a great solution to implement.

But this does not solve all cases: as you realised, the problem here is to control **external** libraries output. And that's not easy. The fix we implement for instance [taken from here](https://stackoverflow.com/questions/5081657/how-do-i-prevent-a-c-shared-library-to-print-on-stdout-in-python/14797594#14797594) is quite convoluted. I recall for instance also having some similar problems in other circumstances (reading miniSEEd maybe?) but I did not find any easy flag in that case, as in this one. So if there is anything I can do to help adding flags also other parts of code with these kind of issues (maybe with someone addressing me to the relevant bits of code), I would like to help.

**Comment by krischer** [comment]
There is this external package here which makes this a lot easier: https://github.com/minrk/wurlitzer

But in general I agree - this should be dealt with but sometimes is just really a lot of work for little perceived benefit.

`obspy.io.mseed` uses `libmseed`'s printing callback functions to convert all log output, warnings, and exceptions to their Python equivalents so they can be caught with the normal means - I think this is by far the cleanest solution but not all external libraries we use in ObsPy have that option.

I'm fine with having this particular option here. Evalresp's days in ObsPy are hopefully also numbered once this here is merged: https://github.com/obspy/obspy/pull/2592

**Comment by megies** [comment]
> > I guess you can directly pass this down through `**kwargs` all the way from `Stream.remove_response()`? If yes maybe add a short comment in that docstring
> 
> I personally prefer to be explicit (which incidentally is also one of the - probably overused - Python dogmas ), given that I do not see major drawbacks (except that yes, maybe the parameter name is too long). so I would let the code as it is, if you do not mind

I was on about how this gets passed down from `Trace.remove_response()` nad yes it seems via kwargs, which is fine for me. Maybe it would be good to mention it very briefly in the `Trace.remove_response()` docstring that it talking about how details get passed down to evalresp methods.

**Comment by megies** [comment]
@rizac we have `CatchOutput` in `core/util/misc.py`, not sure about it's current status though, but it should be able to catch most stuff I think.

**Comment by megies** [comment]
Merging now, but as I said @rizac still feel free to add yourself to CONTRIBUTORS file if you want.

**Comment by rizac** [comment]
> Merging now, but as I said @rizac still feel free to add yourself to CONTRIBUTORS file if you want.

I am already there ;-) 
Thanks

**Pull request merged**